src/rope/utils.rs

```diff
@@ -158,6 +158,10 @@ pub(super) fn is_grapheme_boundary(
 
     debug_assert!(byte_offset <= byte_len);
 
+    if byte_len == 0 {
+        return byte_offset == 0;
```

Comment:
We already know that `byte_offset <= byte_len`.

```suggestion
        return true;
```